lisa/runner.py

```diff
@@ -283,7 +287,7 @@ def _start_loop(self) -> None:
                 )
 
                 for runner in remaining_runners:
-                    while not runner.is_done:
+                    while not runner.is_done and has_idle_worker:
```

Comment:
This change is actually pretty complex, please add what's problem today, and what's improved in the commit message.